nemo/collections/nlp/models/language_modeling/megatron_base_prompt_learning_model.py

```diff
@@ -128,9 +128,9 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         self.pad_token_id = self.tokenizer.pad_id if self.tokenizer.pad_id is not None else self.tokenizer.unk_id
         self.decoder_seq_length = cfg.get('decoder_seq_length', 40)
 
-        if self.trainer.precision == 32:
+        if int(self.trainer.precision) == 32:
```

Comment:
"bf16" will break this unless you check it before so that the `elif` ensures it's a valid int string